container/container.go

```diff
@@ -792,7 +792,9 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 
 	defaultNetName := runconfig.DefaultDaemonNetworkMode().NetworkName()
 
-	if n.Name() == defaultNetName || container.NetworkSettings.IsAnonymousEndpoint {
+	if !container.EnableServiceDiscoveryOnDefaultNetwork() &&
+		(n.Name() == defaultNetName ||
+			container.NetworkSettings.IsAnonymousEndpoint) {
```

Comment:
It would be good to honor `container.NetworkSettings.IsAnonymousEndpoint` regardless if windows supports SD on default network. That will make it consistent with all platforms for all networks.